cmd/nerdctl/pull_linux_test.go

```diff
@@ -0,0 +1,121 @@
+
+	"github.com/containerd/nerdctl/pkg/testutil"
+	"gotest.tools/v3/assert"
+)
+
+type cosignKeyPair struct {
+	publicKey  string
+	privateKey string
+	cleanup    func()
+}
+
+func newCosignKeyPair(t testing.TB, path string) *cosignKeyPair {
+	td, err := os.MkdirTemp(t.TempDir(), path)
+	assert.NilError(t, err)
+
+	cmd := exec.Command("cosign", "generate-key-pair")
+	cmd.Dir = td
+	if out, err := cmd.CombinedOutput(); err != nil {
+		t.Fatalf("failed to run %v: %v (%q)", cmd.Args, err, string(out))
+	}
+
+	publicKey := filepath.Join(td, "cosign.pub")
+	privateKey := filepath.Join(td, "cosign.key")
+
+	return &cosignKeyPair{
+		publicKey:  publicKey,
+		privateKey: privateKey,
+		cleanup: func() {
+			_ = os.RemoveAll(td)
+		},
+	}
+}
+
+func TestImageVerifyWithCosign(t *testing.T) {
+	if _, err := exec.LookPath("cosign"); err != nil {
+		t.Skip()
+	}
+	testutil.DockerIncompatible(t)
+	t.Setenv("COSIGN_PASSWORD", "1")
+	keyPair := newCosignKeyPair(t, "cosign-key-pair")
+	defer keyPair.cleanup()
+	base := testutil.NewBase(t)
+	reg := newTestRegistry(base, "test-image-cosign")
+	defer reg.cleanup()
+	localhostIP := "127.0.0.1"
+	t.Logf("localhost IP=%q", localhostIP)
+	testImageRef := fmt.Sprintf("%s:%d/test-push-signed-image",
+		localhostIP, reg.listenPort)
+	t.Logf("testImageRef=%q", testImageRef)
+
+	dockerfile := fmt.Sprintf(`FROM %s
+CMD ["echo", "nerdctl-build-test-string"]
+	`, testutil.CommonImage)
+
+	buildCtx, err := createBuildContext(dockerfile)
+	assert.NilError(t, err)
+	defer os.RemoveAll(buildCtx)
+
+	base.Cmd("build", "-t", testImageRef, buildCtx).AssertOK()
+	base.Cmd("push", testImageRef, "--sign=cosign", "--cosign-key="+keyPair.privateKey).AssertOK()
+	base.Cmd("pull", testImageRef, "--verify=cosign", "--cosign-key="+keyPair.publicKey).AssertOK()
+}
+
+func TestImageVerifyWithCosignShouldFailWhenKeyIsNotCorrect(t *testing.T) {
+	if _, err := exec.LookPath("cosign"); err != nil {
+		t.Skip()
+	}
+	testutil.DockerIncompatible(t)
+	t.Setenv("COSIGN_PASSWORD", "1")
+	keyPair := newCosignKeyPair(t, "cosign-key-pair")
+	defer keyPair.cleanup()
+	base := testutil.NewBase(t)
+	reg := newTestRegistry(base, "test-image-cosign")
+	defer reg.cleanup()
+	localhostIP := "127.0.0.1"
+	t.Logf("localhost IP=%q", localhostIP)
+	testImageRef := fmt.Sprintf("%s:%d/test-push-signed-image-wrong",
+		localhostIP, reg.listenPort)
+	t.Logf("testImageRef=%q", testImageRef)
+
+	dockerfile := fmt.Sprintf(`FROM %s
+CMD ["echo", "nerdctl-build-test-string"]
+	`, testutil.CommonImage)
+
+	buildCtx, err := createBuildContext(dockerfile)
+	assert.NilError(t, err)
+	defer os.RemoveAll(buildCtx)
+
+	base.Cmd("build", "-t", testImageRef, buildCtx).AssertOK()
+	base.Cmd("push", testImageRef, "--sign=cosign", "--cosign-key="+keyPair.privateKey).AssertOK()
+
+	t.Setenv("COSIGN_PASSWORD", "2")
+
+	newKeyPair := newCosignKeyPair(t, "cosign-key-pair-test")
+
+	base.Cmd("pull", testImageRef, "--verify=cosign", "--cosign-key="+newKeyPair.publicKey).AssertFail()
```

Comment:
can you add an `AssertOK()` test case for pull  plz ?

Reply:
fixed